### ISSUE TITLE
[no-unnecessary-condition] false positive on && operator

### ISSUE THREAD

**Issue by avalanche1** [typescript-eslint/typescript-eslint#2110]
**Repro**

```TS
const bool = false
const res = bool && 123
```

**Expected Result**
no errors
**Actual Result**
on `bool && 123`: `Unnecessary conditional, value is always falsy`
**Additional Info**
`bool` can be true or false in any given moment. why did it decide, that the value is always falsy??

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^3.0.1` |
| `@typescript-eslint/parser`        | `^3.0.1` |
| `TypeScript`                       | `^3.9.3` |
| `ESLint`                           | `^7.0.0` |

**Comment by avalanche1** [comment]
obsolete.
I had outdated "@typescript-eslint/eslint-plugin" version
